site/public/js/gradeable-notebook.js

```diff
@@ -94,43 +130,54 @@ function saveNotebookToLocal() {
  */
 function restoreNotebookFromLocal() {
     if (typeof autosaveEnabled !== 'undefined' && autosaveEnabled) {
-        const state = JSON.parse(localStorage.getItem(notebookAutosaveKey()));
+        const inputs = JSON.parse(localStorage.getItem(notebookAutosaveKey()));
 
-        if (state === null) {
+        if (inputs === null) {
             return;
         }
-
-        // First, we restore multiple choice answers
-        for (const id in state.multiple_choice) {
-            const values = state.multiple_choice[id];
-            // Extract the index from the ID generated from gatherInputAnswersByType()
-            // Index is stored after multiple_choice_, so substring it out
-            const index = id.substring('multiple_choice_'.length);
-            $(`#mc_field_${index} :input`).each((_index, element) => {
-                $(element).prop('checked', values.includes(element.value)).change();
+        //to prevent data loss when these changes get installed on production
+        const not_found = [];
+
+        //restore multiple choice
+        for (const id in inputs.multiple_choice) {
+            const filename = inputs.multiple_choice[id][0];
+            const answers = inputs.multiple_choice[id][1];
```

Comment:
You can also change this line to use [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) to extract the properties from the object (change {} to [] if using array) (totally not necessary though, just something cool)
```suggestion
            const {filename, answers} = inputs.multiple_choice[id];
```